kite-data/kite-data-core/src/main/java/org/kitesdk/data/Flushable.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2013 Cloudera Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kitesdk.data;
+
+/**
+ * A writer that can guarantee data is persisted to disk.
+ * <p>
+ * {@link #flush} a stronger guarantee than {@link java.io.Flushable#flush}.
+ * When it returns, data written is persisted to disk, not just an underlying
+ * buffer.
+ */
+@edu.umd.cs.findbugs.annotations.SuppressWarnings(
+    value="NM_SAME_SIMPLE_NAME_AS_INTERFACE",
+    justification="Intended to be a stricter version of the parent interface")
+public interface Flushable extends java.io.Flushable {
+  /**
+   * Ensure that data has been written to disk.
+   * <p>
+   * This method has a stronger guarantee than {@link java.io.Flushable#flush}.
+   * When this method returns, data written is persisted to disk, not just an
+   * underlying buffer.
+   */
+  @Override
+  void flush();
```

Comment:
Do we really need to consolidate `flush()` and `sync()` semantics into a single method?

In the context of Hadoop, `flush()` (technically `hflush()`) guarantees that data has been flushed from the client and sent to buffers on `${dfs.replication}` number of nodes. The `sync()` (really `hsync()`) method goes further and ensures that the data has been written to stable storage on the remote nodes. Some use cases only need the weaker `hflush()` style semantics and don't want to pay the performance penalty of a fully `hsync()`.